### PR TITLE
modify the FED IDs for phase1 in PixelUnpackingRegion

### PR DESCRIPTION
**Pull request by dkotlins**
PixelUnpackingRegion had some hardwired FED IDs which were incompatible with phase1.
I have updated them.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: EventFilter/SiPixelRawToDigi/src/PixelUnpackingRegions.cc [hunk 1]
```diff
@@ -21,22 +21,22 @@
...
-inline std::ostream& operator<<(std::ostream& s, const PixelUnpackingRegions::Module& m)
-{
-  s<< (isBPIXModule(m.id) ? "BPIX " : "FPIX ") <<m.id<<" "<<m.fed<<"   "<<m.phi<<"   "<<m.x<<" "<<m.y<<" "<<m.z<<"  "<<sqrt(std::pow(m.x,2)+std::pow(m.y,2));
-  return s;
-}
+namespace {
+  bool isBPIXFED(unsigned int fed) {return fed< 32;}
+  bool isFPIXFED(unsigned int fed) {return fed>=32;}
+  //bool isBPIXFEDPhase1(unsigned int fed) {return fed< 1276;}
+  //bool isFPIXFEDPhase1(unsigned int fed) {return fed>=1276;}
```
Comment (slava77) [inline EventFilter/SiPixelRawToDigi/src/PixelUnpackingRegions.cc]: is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #16333 343c6ea2ceb44694bd6364748892f8c2df487a71

Allow for the additional Phase 1 FEDs in PixelUnpackingRegion. There should be no change in monitored quantities.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-10-24-1100 show no significant differences, as expected.

**Comment by davidlange6** [comment]
Hi @dkotlins - two comments
a) please use constants in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/FEDRawData/interface/FEDNumbering.h instead of hardwiring the fed id numbers

b) I think by making the range of fed ids that are expected a member data, the yucky "if phase1" pieces of this code could be isolated to the initialization (where you set isphase1 nicely using the geometry) and no penalty would be paid during the job execution. Could you consider that if not already considered.

**Comment by dkotlins** [comment]
Actually I am considering getting rid of all of this.
It is only used fro printout and one assert.
It will everything simpler.

One can take the min and max FED-ID from the header files but not the split between bpix and fpix FEDs.
This is not stored anywhere.  This is why I did not bother with the FEDnumbering.h.’
But if we get rid of the printout that it will not be needed.

D.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
Thanks! - @slava77, I think this one is ready to go

**Comment by cvuosalo** [comment]
+1

For #16333 63f187e4c763970df8ab9aba69892c3a1207e5a1

Second approval, after minor code clean-up. New Jenkins test results are still OK.

**Comment by davidlange6** [comment]
+1

**Pull request merged**